### DISCUSSION TITLE
Can't Modify Roast Events on MacOS Monterey

### DISCUSSION THREAD

**Discussion by williebonney** [artisan-roaster-scope/artisan#1365] [Q&A] [answered]
When I open up a roast profile, then go to Roast > Properties > Events tab and make changes to any of the pre-configured events at the top (FC Start, Drop, etc), when I click OK, the popup goes away but nothing is updated.  And when I open back up Roast > Properties > Events tab, those value are all reverted back to what they were before I updated them.

Note that I did this roast without Artisan initially.  Then afterwards I started the roast in Artisan, spent about 30 seconds clicking the various buttons to mark the various phases of the roast, and ended the roast.  And now I'm trying to go in and edit those values.

This is on MacOS Monterey v12.4 running Artisan 2.10, though I've also tried this on 2.8.4, 2.8.4 legacy, 2.8.0, and even 2.10.1 and had the same experience in all of them.

However this is my first ever roast, so I'll admit up front that maybe I'm doing something wrong, but I can't figure out what it could be.  Any help is super appreciated!

**Comment by roasterdave** [comment]
`Note that I did this roast without Artisan initially. Then afterwards I started the roast in Artisan, spent about 30 seconds clicking the various buttons to mark the various phases of the roast, and ended the roast. `
I do not understand this process. 

`When I open up a roast profile, then go to Roast > Properties > Events tab and make changes to any of the pre-configured events at the top (FC Start, Drop, etc), when I click OK, the popup goes away but nothing is updated. And when I open back up Roast > Properties > Events tab, those value are all reverted back to what they were before I updated them.`
I can't duplicate this problem.  So we can take a look, please attach your profile *.alog file and your *.aset settings file (Help>> Save Settings).  Rename each file to end in *.txt or zip them before you drag-drop them here on GitHub to a comment box. A reply by email will not work.

**Reply by williebonney** [reply]
Hi Dave.  Thanks for looking into this.

To clarify the process:
1. While roasting the coffee, I took notes in a physical notebook.  I jotted down FC time, the fan and heat changes I made on my Fresh Roast, etc.  The roast took 15 minutes.

2. I opened up Artisan and clicked "Start."

3. I didn't want to spend 15 minutes pretending I was roasting and clicking the buttons at the actual times.  So instead I clicked Charge, Dry End, FC Start, FC End, and Drop (and also the custom buttons I added for Fan and Heat for the Fresh Roast) one after another in quick succession.  Just so I could add events for everything, thinking I could update their times to the correct numbers later.  So it took 38 seconds to click all of those buttons. 

Looks like some of the times have shifted a bit from me messing with it it, but for example currently FC Start, FC End, DROP, and COOL all are marked as 00:38.  And my Fan and Power changes happen at 00:06, 00:08, 00:10, 00:11, etc.

Here's a screenshot of my Roast Properties window so you can see:
<img width="642" alt="roast-properties" src="https://github.com/artisan-roaster-scope/artisan/assets/154944054/4da0d5dd-dea1-4ee1-9402-ea93e0ee0b9d">

4. Then after that 38 seconds, I clicked "Off" to stop the simulated roast.

5. I then saved the profile, and then went to Roast > Properties and clicked the Events tab.   Changed Dry End from 00:38 to 05:00, changed FC Start from 00:38 to 08:30, etc.

6. Then clicked the "OK" button to apply those changes.  But they didn't apply.  So going back to Roast > Properties > Events tab shows them still at 00:38.


Here are the files:
[profile.txt](https://github.com/artisan-roaster-scope/artisan/files/13781449/profile.txt)
[artisan-settings.txt](https://github.com/artisan-roaster-scope/artisan/files/13781451/artisan-settings.txt)

**Comment by roasterdave** [comment, marked as answer]
Thanks for the additional information.  What you are doing to "create" a roast is not going to give what you are looking for.  It is not possible to set events to happen after the end of the roast, the event gets set to 00:38 because that is the closest time to the value you entered.  

Even if you do let Artisan run for 15 minutes there will be no meaningful temperature curves because no valid roast temperature data was recorded.  You could enter your events but without temperature curves it is probably not worth the effort.  Artisan is intended to make live recordings with attached temperature instrumentation. 

However, it is possible to create a roast profile from a set of notes.  Set the **Meter** (Config>> Device>> ET/BT tab) to **NONE**.  After you press START hit the SPACE BAR to enter temperatures and click the EVENT buttons at the appropriate times to enter those events.  It will take the same amount of time as the entire roast.  Alternately, you could create a CSV file to import or use the Designer to create a curve based upon your temperature points.  

I recommend your time will be better spent instrumenting your roaster so you can capture the full fidelity of your roast data with Artisan.

**Reply by williebonney** [reply]
Awesome, thank you so much for the info.  I have a probe on the way, but it's not here yet and I was eager to do an initial roast.  I figured if I could record the timings from this first roast in Artisan, then I'd have all my notes in one place and could use it to guide me once I get the probe set up.

I'll try out creating the roast profile from my notes via the method you suggested.

I appreciate all of your help!
